motor/core.py

```diff
@@ -448,7 +448,7 @@ def __init__(self, client, name, **kwargs):
         delegate = kwargs.get('_delegate') or Database(
             client.delegate, name, **kwargs)
 
-        super(self.__class__, self).__init__(delegate)
+        super(AgnosticBaseProperties, self).__init__(delegate)
```

Comment:
This should be `AgnosticDatabase` instead of `AgnosticBaseProperties`.

Reply:
This does not work because at run time self is an instance of `motor.motor_asyncio.AsyncIOMotorDatabase` which is not a child class of `AgnosticDatabase`, hence the previous use of `self.__class__`
I don't see a problem using `super(AgnosticBaseProperties, self).__init__(delegate)` since thats exactly what we are trying to invoke here, right?

Reply:
Yes that's a great point. I understand the problem now. Still, using the parent class in these super calls worries me a bit because the usual pattern is to use the current class.  We're currently dropping support for Python 2 (https://github.com/mongodb/motor/pull/61). After dropping 2, I think it would be possible to use "bare" super() calls everywhere. 

Could you try using `super()` instead and ignore the Python 2 test failures? It would also be great if you could add a test for your use-case of extending these classes.